tests/unit/test_postgresql_provider.py

```diff
@@ -153,12 +174,37 @@ def test_on_database_requested(
             self.request_database()
             self.assertTrue(isinstance(self.harness.model.unit.status, BlockedStatus))
 
+    def test_on_relation_departed(self):
+        # Workaround ops.testing not setting `departing_unit` in v1.5.0
+        # ref https://github.com/canonical/operator/pull/790
+        with patch.object(
+            RelationDepartedEvent, "departing_unit", new_callable=PropertyMock
+        ) as mock_departing_unit:
+            # Test that no flag is set in peer relation data if the remote unit
+            # is the unit that is being removed.
+            mock_departing_unit.return_value = self.harness.model.get_unit("application/0")
+            self.harness.remove_relation_unit(self.rel_id, "application/0")
+            peer_relation_data = self.harness.get_relation_data(self.peer_rel_id, self.unit)
+            self.assertDictEqual(peer_relation_data, {})
+
+            # Test that a flag for preventing user deletion is set in peer relation data
```

Comment:
quick question do you have an integration test for verifying that your "flag preventing user deletion" works?

Reply:
Yes Mia! It is [here](https://github.com/canonical/postgresql-operator/blob/d139e248a5b490f442b0d9d1ecc10f7516cbd56a/tests/integration/test_db.py#L114) (part of the other PR: https://github.com/canonical/postgresql-operator/pull/18). When the two units are destroyed in that line, the deletion of the user would be triggered in them even we still have two more units in the relation with the application. So, with the flag, when the two units are destroyed, the user is not deleted (it is deleted when the relation is really broken between the database and application).

Reply:
Awesome!!